editor/state.js

```diff
@@ -131,6 +131,25 @@ export function hoveredBlock( state = null, action ) {
 }
 
 /**
+ * Reducer returning the focused block state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export function focus( state = {}, action ) {
```

Comment:
Tried this and it has several implications, we should ignore consecutive focuses etc... I think this should be added but in its own PR.